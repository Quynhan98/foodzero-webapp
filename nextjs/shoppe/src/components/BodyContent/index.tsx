import { ChangeEvent, memo, ReactNode } from 'react'
import { Box, Heading, Flex } from '@chakra-ui/react'

// Components
import Search from '@components/Search'

interface BodyContentProps {
  onChange: (e: ChangeEvent<HTMLInputElement>) => void
  onKeyDown: (e: React.KeyboardEvent<HTMLInputElement>) => void
  onClick: (e: React.MouseEvent<HTMLButtonElement>) => void
  children: ReactNode
}
const BodyContent = ({
  onChange,
  onKeyDown,
  onClick,
  children,
}: BodyContentProps) => {
  return (
    <Box pt="96px">
      <Heading as="h2" fontSize="large" fontWeight="medium">
        Shop The Latest
      </Heading>
      <Flex pt="40px" gap="35px" justifyContent="space-between">
        <Box minW="261px">
          <Search
            onChange={onChange}
            onKeyDown={onKeyDown}
            onClick={onClick}
            placeholder="Search..."
          />
        </Box>
        <Flex
          flexDirection="column"
          justifyContent="center"
          paddingBottom="50px"
          margin="0 auto"
        >
          {children}
        </Flex>
      </Flex>
    </Box>
  )
}

export default memo(BodyContent)
