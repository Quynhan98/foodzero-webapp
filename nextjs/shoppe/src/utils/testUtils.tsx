import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>{component}</ChakraProvider>,
  )
}

export * from '@testing-library/react'
export { customRender as render }
