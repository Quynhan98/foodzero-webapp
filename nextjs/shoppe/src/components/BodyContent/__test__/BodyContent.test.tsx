// Utils
import { fireEvent, render, screen } from '@utils/testUtils'

// Components
import BodyContent from '@components/BodyContent'

describe('BodyContent render', () => {
  const props = {
    onChange: jest.fn(),
    onKeyDown: jest.fn(),
    onClick: jest.fn(),
  }

  it('should match data for BodyContent component', () => {
    const { container } = render(
      <BodyContent {...props}>List Data</BodyContent>,
    )

    expect(container).toMatchSnapshot()
  })

  it('should call onChange Search ', () => {
    render(<BodyContent {...props}>List Data</BodyContent>)

    const search = screen.getByRole('textbox') as HTMLInputElement

    fireEvent.change(search, {
      target: { value: 'Search Value' },
    })

    expect(props.onChange).toBeCalled()
    expect(search.value).toBe('Search Value')
  })

  it('should call onClick Icon Search ', () => {
    render(<BodyContent {...props}>List Data</BodyContent>)

    const iconSearch = screen.getByRole('button')

    fireEvent.click(iconSearch)
    expect(props.onClick).toBeCalled()
  })

  it('should call onKeyDown after Enter ', () => {
    render(<BodyContent {...props}>List Data</BodyContent>)

    const search = screen.getByRole('textbox')
    fireEvent.keyDown(search)

    expect(props.onKeyDown).toBeCalled()
  })
})
