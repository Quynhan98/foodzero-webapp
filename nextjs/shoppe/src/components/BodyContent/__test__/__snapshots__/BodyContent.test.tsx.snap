// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BodyContent render should match data for BodyContent component 1`] = `
<div>
  <div
    class="css-butksl"
  >
    <h2
      class="chakra-heading css-aj5n6i"
    >
      Shop The Latest
    </h2>
    <div
      class="css-17dci3b"
    >
      <div
        class="css-1i5wex2"
      >
        <div
          class="css-zrbfnb"
        >
          <input
            class="chakra-input css-b2ycb2"
            placeholder="Search..."
            type="text"
          />
          <button
            aria-label="Search database"
            class="chakra-button css-1ljo2n9"
            type="submit"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-bokek7"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="css-aozu9f"
      >
        List Data
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
